model/task_lifecycle.go

```diff
@@ -1570,24 +1570,32 @@ func resetStrandedTask(t *task.Task) error {
 		// If the task has already exceeded the unschedulable threshold, we
 		// don't want to restart it, so just mark it as finished.
 		if t.DisplayOnly {
-			for _, etID := range t.ExecutionTasks {
-				var execTask *task.Task
-				execTask, err := task.FindOneId(etID)
-				if err != nil {
+			var execTasks []task.Task
+			var err error
+
+			if t.ResetFailedWhenFinished && !t.ResetWhenFinished {
+				execTasks, err = task.Find(task.FailedTasksByIds(t.ExecutionTasks))
+			} else {
+				execTasks, err = task.FindAll(db.Query(task.ByIds(t.ExecutionTasks)))
+			}
+
+			if err != nil {
+				return errors.Wrap(err, "finding execution tasks")
+			}
+
+			for _, execTask := range execTasks {
+				if err != nil { // Keeping existing error check logic?
 					return errors.Wrap(err, "finding execution task")
 				}
-				if execTask == nil {
```

Comment:
should we still check for empty exec tasks? 
like len(execTasks) = 0?

Reply:
I added a test to see if it's 0 and to error out. But i realized that this method is often given the exec task as the running host and not the display task so I don't know if this is the right logic to add here